src/mango/src/mango_native_proc.erl

```diff
@@ -150,6 +148,22 @@ get_index_entries({IdxProps}, Doc) ->
     end.
 
 
+get_index_values(Fields, Selector, Doc) ->
+    lists:map(fun({Field, _Dir}) ->
+        case mango_doc:get_field(Doc, Field) of
+            not_found -> 
+                not_found;
+            bad_path -> 
+                not_found;
+            Value -> 
+                case should_index(Selector, Doc) of
```

Comment:
this looks like it could be more efficient - we only need to check whether the doc should be indexed once and, if not, return immediately.